### PR TITLE
Robot list frontend

### PR DESCRIPTION
**Pull request by mdavis60**
Added ability to view all devices in the table on the home screen.  Clicking on an available device assigns the current user to the device.  Clicking on a device that the current user is connected to will release that device.

### CODE REVIEW

File: frontEnd/src/js/react-components/App.tsx [hunk 1]
```diff
@@ -1,14 +1,18 @@
 import * as React from 'react';
 import {IServerController} from "../models/ServerController";
 import {ICookies} from "../models/Cookies";
+import {DeviceTable} from "./devices/DeviceTable";
+import {DeviceController, IDeviceController} from "../models/devices/DeviceController";
+import Timer = NodeJS.Timer;
```
Comment (thomassimko) [inline frontEnd/src/js/react-components/App.tsx]: I don't think we need this anymore

File: frontEnd/src/js/react-components/App.tsx [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 
 export interface IAppProps {
     serverController: IServerController
+    deviceController: IDeviceController
     cookies:ICookies
```
Comment (thomassimko) [inline frontEnd/src/js/react-components/App.tsx]: Can delete this and the serverController now since we don't use it.

Comment (thomassimko) [reply frontEnd/src/js/react-components/App.tsx]: Remove their imports as well

File: frontEnd/src/js/react-components/devices/DeviceTable.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                    }
+                }}
+                defaultPageSize={10}
+                className="-striped -highlight"
+            />
+            <br />
+        </div>
+    }
+
+    private async initializeDeviceList(){
```
Comment (thomassimko) [inline frontEnd/src/js/react-components/devices/DeviceTable.tsx]: space after ()

File: webServer/src/test/java/com/jnicho/quickpick/DeviceTests.java [hunk 4]
```diff
@@ -185,7 +185,7 @@ public void testReleaseByOtherUser() throws Exception {
         User user = userRepository.findOne(1L);
         device.setConnectedUser(user);
         deviceRepository.save(device);
-        mvc.perform(post("/api/devices/1/release"))
+        mvc.perform(get("/api/devices/1/release"))
```
Comment (mdavis60) [inline webServer/src/test/java/com/jnicho/quickpick/DeviceTests.java]: I changed all these to gets.  I'm pretty sure they should technically be puts, but a put request isn't implemented on the frontend yet.  I can make a ticket to fix these next sprint.

File: frontEnd/src/js/react-components/App.tsx (hunk 1 above)
Comment (mdavis60) [reply frontEnd/src/js/react-components/App.tsx]: The bottom two imports?
we definitely need DeviceTable

Comment (thomassimko) [reply frontEnd/src/js/react-components/App.tsx]: import Timer

File: frontEnd/src/js/react-components/App.tsx [hunk 5]
```diff
@@ -1,14 +1,15 @@
 import * as React from 'react';
-import {IServerController} from "../models/ServerController";
 import {ICookies} from "../models/Cookies";
+import {DeviceTable} from "./devices/DeviceTable";
+import {DeviceController, IDeviceController} from "../models/devices/DeviceController";
 
 export interface IAppProps {
-    serverController: IServerController
+    deviceController: IDeviceController
     cookies:ICookies
```
Comment (thomassimko) [inline frontEnd/src/js/react-components/App.tsx]: You forgot about the cookies! and their import

**Pull request merged**